packages/crayons-core/src/components/datepicker/datepicker.tsx

```diff
@@ -81,6 +81,52 @@ const getWeekDays = (lang): any => {
     format(addDays(startOfWeek(new Date()), i), 'EEEEE', { locale: lang })
   );
 };
+
+const parseIcelandicDate = (value, langModule) => {
+  // For Icelandic language, the date format is different. There is a discrepency which is handled in this PR https://github.com/date-fns/date-fns/pull/3934
+  if (!value) return value;
+  const icelandicLanguageDisplayFormat = 'dd MMMM yyyy';
+  const icelandicMonthMapper = {
+    'jan.': 'jan.',
+    'feb.': 'feb.',
+    'mars': 'm',
+    'apríl': 'apríl',
+    'maí': 'maí',
+    'júní': 'júní',
+    'júlí': 'júlí',
+    'ágúst': 'á',
+    'sept.': 's',
+    'okt.': 'ó',
+    'nóv.': 'n',
+    'des.': 'd',
+  };
+  const correctedDate = value?.replace(
+    /jan\.|feb\.|mars|apríl|maí|júní|júlí|ágúst|sept\.|okt\.|nóv\.|des\./g,
+    (match) => icelandicMonthMapper[match]
+  );
+  return parseDate(
+    correctedDate,
+    icelandicLanguageDisplayFormat,
+    new Date(),
+    langModule
+  );
+};
+
+const parse = (value, displayFormat, date, langModule) => {
+  if (!value) return value;
+  if (langModule?.locale?.code === 'is' && displayFormat === 'dd MMM yyyy') {
+    return parseIcelandicDate(value, langModule);
+  }
+  return parseDate(value, displayFormat, date, langModule);
+};
+
+const isMatch = (value, displayFormat, langModule) => {
+  return (
```

Comment:
if langmodule is `is`, this will return false. is this the expectation?

Reply:
Yes @arvindanta , as isMatch is not supported completely for icelandic language dates.
Though, we are parsing through mapping, the isMatch method would still throw error for the same